swift/extractor/visitors/VisitorBase.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <swift/AST/ASTVisitor.h>
+#include <swift/AST/TypeVisitor.h>
+
+#include "swift/extractor/SwiftDispatcher.h"
+
+namespace codeql {
+
+namespace detail {
+class DispatcherWrapper {
```

Comment:
First it seemed to me that this just created an extra level of indirection around `SwiftDispatcher` and didn't add any functionality. However, this just seems to be the shared base class for `AstVisitorBase` and `TypeVisitorBase`. Wouldn't `VisitorBase` be a more appropriate name. Also If not other `VisitorBase`s will be added in the future and if no methods are expected to be added to this class, it's probably perfectly fine to just introduce a dispatcher field in both `VisitorBase`s?

Reply:
For the first point I'm perfectly fine with renaming this `VisitorBase`. For the second, we already share the constructor this way, and I already have some draft PRs adding shared methods here, so I would leave the shared base class.

Reply:
If methods are going to be added, then definitely having a class here is useful.

Reply:
So just rename.